docs/rfcs/sync-manager.md

```diff
@@ -0,0 +1,171 @@
+
+* Query info/collections and pass this information to engines.
+
+* Manage interaction with the token server to obtain a token. As part of this, the manager needs to detect when an engine fails to use the token, generally due to a node reassignment mid sync. In this scenario, the manager will abort any pending syncs and reset itself, starting a new sync flow. Note also that this means the containing application will generally not need to see authentication errors from engines, only from the manager.
+
+* Manage the "clients" collection - we probably can't ignore this any longer, especially for bookmarks (as desktop will send a wipe command on bookmark restore). This involves both communicating with the engine regarding commands targetting it, and accepting commands to be send to other devices. Note however that it's not yet clear where these commands will originate from - eg, consider a bookmark restore - which point in this process will queue the wipe commands for other clients?). Exactly how to manage this (eg, when we can assume the engine has processed the command?) are TBD.
+
+* Perform, or coordinate, the actual sync of the engines - from the containing app's POV, there's a single "sync now" entry-point. Exactly how the manager binds to engines is TBD. For example, a possible implementation is that a callback is registered with the sync manager. It's also likely that we will expose a way to sync a specific engine rather than "sync all".
+
+* Manage the fact that there may not be a 1:1 relationship between collections and sync engines. For example, it may be true in the future that a single engine uses multiple collections. A possible example is when we come to syncing "containers" - the history engine may then use 2 collections as part of a sync.
+
+## What the sync manager will not do
+
+For completeness, this section lists things which the sync manager will not do:
+
+* It is not a sync scheduler. The containing app will be responsible for knowing when to sync (but as above, the scheduler does know *what* to sync.
+
+(anything else?)
+
+# Current implementations and challenges with the Rust components
+
+In most current sync implementations, all engines and management of this global is located in the "sync" code - they are very close to each other and tightly bound. However, the rust components have a different structure which offers some challenges.
+
+* Some apps only care about a subset of the engines - lockbox is one such app and only cares about a single collection/engine.
+
+* Some apps will use a combination of Rust components and "traditional" engines. For example, iOS is moving some of the engines to using Rust components, which other engines are likely to never be ported. We also plan to introduce some rust components into desktop in the same way, meaning desktop will eventually have both rust components and "traditional" engines involved.
```

Comment:
Which engines on iOS aren't likely to be ported (Possibly worth calling out, just for completeness)

Reply:
> Which engines on iOS aren't likely to be ported (Possibly worth calling out, just for completeness)

That's a great question for the long term - but over the next few months, I suspect we will consider things a complete success if the only engines which are ported are bookmarks and logins.